src/main/java/htsjdk/samtools/cram/build/CramContainerHeaderIterator.java

```diff
@@ -0,0 +1,85 @@
+package htsjdk.samtools.cram.build;
+
+import htsjdk.samtools.cram.io.InputStreamUtils;
+import htsjdk.samtools.cram.structure.Container;
+import htsjdk.samtools.cram.structure.ContainerIO;
+import htsjdk.samtools.cram.structure.CramHeader;
+import htsjdk.samtools.seekablestream.SeekableStream;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+/**
+ * Iterate over CRAM containers from an input stream, and unlike {@link CramContainerIterator} only
+ * the header of each container is read, rather than the whole stream. As a result, the container block
+ * data is *not* populated, including the compression header block and slices.
+ *
+ * This class is useful when you are not interested in the contents of containers, for example when indexing container
+ * start positions.
+ */
+public class CramContainerHeaderIterator implements Iterator<Container> {
+  private CramHeader cramHeader;
+  private SeekableStream inputStream;
+  private Container nextContainer;
+  private boolean eof = false;
+  private long offset = 0;
+
+  public CramContainerHeaderIterator(final SeekableStream inputStream) throws IOException {
+    cramHeader = CramIO.readCramHeader(inputStream);
+    offset = inputStream.position();
```

Comment:
In looking at this I noticed that the existing `CramContainerIterator` doesn't correctly set the offset as you're ding here (it fails to account for the size of the Cram header, so the first real container has it's offset set to 0 - all of them are off after that). If we fix that, then this class can be a subclass of that one, and we won't have to duplicate the rest of the implementation. Plus we won't have to artificially constrain the stream to be a `SeekableStream`.

Reply:
I wanted to explore this a bit since I'm a little surprised the existing `CramContainerIterator` doesn't cause problems due to the bad offsets. Anyway, here is a [commit](https://github.com/cmnbroad/htsjdk/commit/d128c26f7ed6fae425380a515da3ce56655e2d24) with my proposed fix - see what you think.

Reply:
Your fix is a great improvement - it simplifies the new code and fixes a previous issue with the code. Thank you!

Reply:
BTW I had a quick look at why this wasn't detected before. I changed the code in Squark that uses CramContainerHeaderIterator to use CramContainerIterator - and for small CRAM files it still works, most likely because a Chunk that starts at the start of the file, rather than the first offset, works in the same way. For large CRAM files the offsets don't work however and the code throws an exception.

So it may be that testing on small CRAM files has masked this bug. CRAM indexes written with the old code may be invalid though.